docs/micro-service-architecture.md

```diff
@@ -0,0 +1,248 @@
+**Pros:**
+- Open source
+- Widely use with a strong community
+- Wide range of features
+
+**Cons**
+- Complexity, when first trying keycloak it can be kind of complex to understand it's role and implications
+- Redhat documentation, keycloak documentation might be very hard to get through at some points due to it's documentation.
+
+#### Zitadel
+**Pros**
+- User-Centric Approach: Zitadel is designed with a strong focus on user experience and user-centric identity management.
+- Strong Security Features: Zitadel emphasizes security and includes features such as passwordless authentication and zero-trust architecture.
+- Enterprise-Grade: Zitadel positions itself as an enterprise-grade solution, catering to the needs of larger organizations.
+
+**Cons**
+- Lesser Adoption: As of my last knowledge update in January 2022, Zitadel might have had fewer users and 
+community support compared to more established solutions like Keycloak or Auth0.
+- Pricing: While Zitadel offers a free tier, pricing for larger enterprises might be a consideration.
+
+#### Auth0
+**Pros**
+- Ease of Use: Auth0 is known for its ease of use and quick setup. It provides a user-friendly dashboard and supports various identity scenarios out of the box.
+- Large User Base: Auth0 is widely adopted and has a large user base, leading to a robust community and extensive third-party integrations.
+- Extensive Documentation: Auth0 provides thorough and well-organized documentation, making it easier for developers to implement and troubleshoot.
+
+**Cons**
+- Pricing: Auth0's pricing might be higher compared to some other solutions, especially for large enterprises with high user counts.
+- Limited Customization in Free Tier: Some advanced features and customization options may be restricted in the free tier.
+
+
+## Services
+![microservices](./ms.png)
+
+### `API Gateway`
+
+The API Gateway is a single entry point for all clients. 
+It provides HTTP endpoints to access the different services in the infrastructures, and does so while checking the authorization claims of the user doing the request. 
+This service works closely with the Keycloak service.
+
+
+### `Keycloak`
+
+It will allow us to provide a secure authentication and authorization service for the client. 
+
+### `Notification Service`
+
+The Notification Service is responsible for sending notifications to the users. The goal is to have a single service responsible for sending notifications to the users.
+
+### `Job Service`
+
+The Job offer service is responsible for managing jobs, job offers and companies. This service is implemented in kotlin using spring r2dbc as a reactive database driver. 
+We chose r2dbc instead of jpa since being we needed to have asynchronous requests in our architecure to optimize latency. 
+Having a synchronous system would mean we can only treat one request at the time which at larger scale will not be efficient enough. 
+The service uses a postgres database to store it's data.
```

Comment:
```suggestion
The Job Service is responsible for managing jobs, job categories, job offers and companies. This service is implemented in Kotlin using Spring R2DBC as a reactive database driver. 

We chose R2DBC instead of JPA because we need to have asynchronous requests in our architecture to optimize latency. 
Having a synchronous system would mean we could only treat one request at a time which, at larger scale, would not be efficient enough.

The service uses a Postgres database to store its data.
```